commerce-service/src/main/java/com/liferay/commerce/internal/messaging/CheckCommerceSubscriptionOrderPayedMessageListener.java

```diff
@@ -119,26 +123,94 @@ protected void doReceive(Message message) throws Exception {
 
 			calendar.setTime(creationDate);
 
+			CommerceOrder commerceOrder =
+				firstCommerceSubscriptionCycleEntry.fetchCommerceOrder();
+
 			calendar.add(
 				Calendar.MINUTE,
-				_commerceSubscriptionConfiguration.payedOrderInterval());
+				_commercePaymentEngine.payedOrderInterval(
+					commerceOrder.getCommerceOrderId()));
 
-			CommerceOrder commerceOrder =
-				firstCommerceSubscriptionCycleEntry.fetchCommerceOrder();
+			boolean checkSubscriptionPayments = _checkSubscriptionPayments(
+				commerceSubscriptionEntry);
 
-			if ((commerceOrder != null) &&
-				!(commerceOrder.getPaymentStatus() ==
-					CommerceOrderConstants.PAYMENT_STATUS_PAID) &&
-				!(DateUtil.compareTo(calendar.getTime(), now) < 0)) {
+			if (!checkSubscriptionPayments ||
+				((commerceOrder != null) &&
+				 !(commerceOrder.getPaymentStatus() ==
+					 CommerceOrderConstants.PAYMENT_STATUS_PAID) &&
+				 !(DateUtil.compareTo(calendar.getTime(), now) < 0))) {
 
-				_commerceSubscriptionEntryLocalService.updateSubscriptionStatus(
-					commerceSubscriptionEntry.getCommerceSubscriptionEntryId(),
-					CommerceSubscriptionEntryConstants.
-						SUBSCRIPTION_STATUS_SUSPENDED);
+				_commercePaymentEngine.suspendSubscription(
+					commerceSubscriptionEntry.getCommerceSubscriptionEntryId());
 			}
 		}
 	}
 
+	private boolean _checkSubscriptionPayments(
+			CommerceSubscriptionEntry commerceSubscriptionEntry)
+		throws Exception {
+
+		CommerceOrderItem oldCommerceOrderItem =
+			_commerceOrderItemLocalService.getCommerceOrderItem(
+				commerceSubscriptionEntry.getCommerceOrderItemId());
+
+		CommerceOrder commerceOrder = oldCommerceOrderItem.getCommerceOrder();
+
+		CommerceSubscriptionStatusResult subscriptionPaymentDetails =
+			_commercePaymentEngine.getSubscriptionPaymentDetails(
+				commerceOrder.getCommerceOrderId());
+
+		if (subscriptionPaymentDetails == null) {
+			return false;
+		}
+
+		long retrievedCyclesCompleted =
+			subscriptionPaymentDetails.getCyclesCompleted();
+		long retrievedCyclesRemaining =
+			subscriptionPaymentDetails.getCyclesRemaining();
+		long paymentsFailed = subscriptionPaymentDetails.getPaymentsFailed();
+
+		int commerceSubscriptionCycleEntriesCount =
+			commerceSubscriptionEntry.
+				getCommerceSubscriptionCycleEntriesCount();
+		long maxSubscriptionCycles =
+			commerceSubscriptionEntry.getMaxSubscriptionCycles();
+
+		long retrievedMaxCycles =
+			retrievedCyclesCompleted + retrievedCyclesRemaining;
+
+		if (maxSubscriptionCycles != retrievedMaxCycles) {
+			return false;
+		}
+
+		if (retrievedCyclesCompleted <
```

Comment:
Hey @luca-pellizzon , if I'm understanding this line right, I don't think we want to do this. If the cycle-length is shorter than the payment waiting period (i.e. 48 hours for PayPal), then there may be more than 1 payment pending. We can talk this over tomorrow in person :)